### DISCUSSION TITLE
Unable to do custom comparison levels with non exact match sql statements

### DISCUSSION THREAD

**Discussion by mmagoffin-sd** [moj-analytical-services/splink#799] [Q&A] [answered]
I was trying to create a custom comparison level for looking at first names by seeing if a canonical name matched with a first name that was a nickname given in my data. However, since I was doing a LIKE statement in SQL, I didn't need a term frequency column. When I try to run linker.estimate_parameters_using_expectation_maximization I get a BinderException about a tf_canonicals_fname_l column that shouldn't need to exist. I tried both a different SQL statement (jaro_winkler_similarity(str_l,str_r) < 0.8) and attempting to add a term frequency column based on my fname column within the comparison. Both the custom comparison level and the error are in screenshots below:

<img width="829" alt="Screen Shot 2022-09-21 at 4 42 45 PM" src="https://user-images.githubusercontent.com/85239124/191606492-7c1ecb7c-eb85-4665-902b-8aa0871c338f.png">

<img width="652" alt="Screen Shot 2022-09-21 at 4 41 35 PM" src="https://user-images.githubusercontent.com/85239124/191606271-55112754-8b52-4b0f-81ff-6421dbb98cda.png">

**Comment by RobinL** [comment, marked as answer]
Sorry for the bug - I think this was fixed in version 3.3.5 [here](https://github.com/moj-analytical-services/splink/releases/tag/v3.3.5) so if you upgrade `pip install --upgrade splink` this problem should go away.  Let me know if it doesn't!

**Reply by mmagoffin-sd** [reply]
Looks like it fixed the problem for running the estimate_parameters_using_expectation_maximization() function, but now I get the same error on linker.predict() 
<img width="872" alt="Screen Shot 2022-09-22 at 10 04 04 AM" src="https://user-images.githubusercontent.com/85239124/191768397-049ce169-7a89-4820-850a-fa9a91025712.png">
Thank you for the quick responses!

**Reply by RobinL** [reply]
Please could you upgrade to 3.3.7 and see if it works.  I've added a test [here](https://github.com/moj-analytical-services/splink/pull/803) with your specific example, so it definitely should work now!

**Reply by mmagoffin-sd** [reply]
That worked, thank you!
